### DISCUSSION TITLE
Shareable and extendible configurations

### DISCUSSION THREAD

**Discussion by Avasam** [microsoft/pyright#5278] [Ideas]
Is shareable configurations something that can be considered?

Most of the tooling I use, both personally and at work, support some form of shareable configs from outside the project (TypeScript, ESLint, dprint, Ruff is on its way, and we're working on a tool to merge editor configurations). This allows us to centralize strict and best practices configurations across the organization easily, propagate updates, as well as reusing those in our personal projects. Whilst still allowing project-specific configs, and individual projects to be less strict if they want to.

Shareable configurations somewhat imply having extendible configurations in the first place. Which would also alleviate some pain points in mono-repo projects (typeshed being one of them) where the editor cannot properly show type-hinting errors for the current folder's configuration.

**Comment by erictraut** [comment]
The configuration mechanism in pyright is already incredibly complex. We need to compose configuration settings from LSP settings (user and workspace), command-line options, pyrightconfig.json files and pyproject.toml files, not to mention per-file overrides. It's difficult for maintainers and users to reason about how all of these compose. I'm reluctant to add yet more complexity unless there's a really strong signal from users that it's needed. So for now, I don't think this is something we'd want to add. It's something we could consider in the future if there are sufficient upvotes.
